pandas/_libs/internals.pyx

```diff
@@ -64,7 +64,8 @@ cdef class BlockPlacement:
 
         return '%s(%r)' % (self.__class__.__name__, v)
 
-    __repr__ = __str__
+    def __repr__(self):
```

Comment:
Not sure if this is a cython bug, but I wasn't getting a nice repr. without this change.

Reply:
usually you would just do
``__repr__ = __str__``

Reply:
Indeed, that's what I deleted. But on master I get this for the repr.

```python
In [1]: import pandas as pd; import numpy as np; import pickle

In [2]: pd.Series([1,2])._data._block.mgr_locs
Out[2]: <pandas._libs.internals.BlockPlacement object at 0x111b65048>
```